### PR TITLE
A4A: Introduce new mixins for consistent typography scale

### PR DESCRIPTION
**Pull request by keoshi**
Related P2 post: pfunGA-1Q2-p2

## Proposed Changes

This PR introduces new SCSS `@mixins` that can be easily used to create a consistent typographic scale across all of A4A. It also updates most A4A touch points to use these new styles.

The introduced styles match those presented in pfunGA-1zi-p2 and in Figma https://www.figma.com/design/Ij0yoCBbL4ZeZffvv1278l/A4A-Library?node-id=6431-138763&t=oTgu09K8vee8VOPd-11

They also match Gutenberg's nomenclature; we've added XXL as a new style.

- **Heading XXL**: `a4a-font-heading-xxl`
- **Heading XL**: `a4a-font-heading-xl`
- **Heading LG**: `a4a-font-heading-lg`
- **Heading MD**: `a4a-font-heading-md`
- **Body LG**: `a4a-font-body-lg`
- **Body MD**: `a4a-font-body-md`
- **Body SM**: `a4a-font-body-sm`
- **Label / Button**: `a4a-font-label-button`

They can be used by either using them wholly, or by overriding specific properties. They all share these properties: `$font-size`, `$font-weight`, `$line-height`. Additionally, Heading XXL, XL and LG support an additional property: `$letter-spacing`.

Usage examples:

- Wholly: `@include a4a-font-heading-xl`
- Overriding one property: `@include a4a-font-body-md($font-weight: 500)`
- Overriding multiple properties: `@include a4a-font-heading-xxl($font-weight: 200; $letter-spacing: 4px)`

image

## Questions

Got a couple of questions for @Automattic/jetpack-genesis:

1. Is it ok to add global mixins to Calypso like this 6c7cc1ff493e641f34791a9a16ec9379a1869dab ? Is it going to affect the package size of Calypso? I couldn't figure out a different way of adding these locally to A4A that would work. CC'ing @sgomes if you have any tips here.
2. For some reason, the mixins mentioned above didn't work in the Marketplace section, so I had to duplicate them here: 8516639144485bd4abc9c9591d10a2e75ddbd7dd — any reason why the global mixins aren't usable here?

## Testing Instructions

* Fire up this PR.
* Go to A4A.
* Make sure all touch points look ok and that nothing is breaking.

## Screenshots

Before | After
--|--
image | image
image | image
image | image
image | image

### CODE REVIEW

**Comment by sgomes** [comment]
Hey @keoshi! 👋

> Is it ok to add global mixins to Calypso like this https://github.com/Automattic/wp-calypso/commit/6c7cc1ff493e641f34791a9a16ec9379a1869dab ? Is it going to affect the package size of Calypso? I couldn't figure out a different way of adding these locally to A4A that would work. CC'ing @sgomes if you have any tips here.

Mixins are an exclusively compile-time construct that gets used to generate CSS, which is to say that they don't actually end up in the generated CSS, nor do they introduce modularization issues through runtime dependencies. They do have the potential to bloat the resulting code when used carelessly (because of combinatorial explosion), but that's no different than most Sass features.

In this case, it appears that there isn't a significant difference between what was there before and what the mixin usage will generate 👍

**Review by madebynoam — APPROVED** [review]
Thank you, @keoshi! 🙌

I tested it on the live link, and reviewed the code.
Lgtm!

**Comment by cleacos** [comment]
I'm going to add the Loop team from Dotcom as a reviewer. We are touching on common styles in the ItemsDashboard and in the Layout.

**Comment by keoshi** [comment]
Thanks for doing that, @cleacos.

Note for myself to fix: the one obvious regression I see here is in the preview panel's title:

Before | After
--|--
image | image

File: client/a8c-for-agencies/components/user-contact-support-modal-form/style.scss [hunk 1]
```diff
@@ -54,9 +50,7 @@
 }
 
 .a4a-contact-support-modal-form__title {
-	font-size: 1.5rem;
-	font-weight: 700;
-	line-height: 0.8;
+	@include a4a-font-heading-xl;
```
Comment (yashwin) [inline client/a8c-for-agencies/components/user-contact-support-modal-form/style.scss:53]: `font-weight` here is 700, and in the `a4a-font-heading-xl` it is 600. Is that ok?

File: client/a8c-for-agencies/components/user-feedback-modal-form/style.scss [hunk 2]
```diff
@@ -44,16 +44,13 @@
 }
 
 .user-feedback-modal-form__title {
-	font-size: 1.5rem;
-	font-weight: 700;
-	line-height: 0.8;
+	@include a4a-font-body-lg;
```
Comment (yashwin) [inline client/a8c-for-agencies/components/user-feedback-modal-form/style.scss:47]: font-weight here is 700, and in the `a4a-font-body-lg` it is 400. Is that ok?

File: client/a8c-for-agencies/sections/marketplace/checkout/style.scss [hunk 3]
```diff
@@ -116,28 +114,23 @@
...
-		font-size: 1.25rem;
-		font-weight: 600;
+		@include a4a-font-heading-lg;
 		color: var(--color-neutral-80);
-		line-height: 1;
 
 		@include break-xlarge {
-			font-size: 1.5rem;
-			font-weight: 700;
+			@include a4a-font-heading-xl;
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/checkout/style.scss:121]: `font-weight` here is 700, and in the `a4a-font-heading-xl` it is 600. Is that ok?

File: client/a8c-for-agencies/sections/marketplace/checkout/style.scss [hunk 4]
```diff
@@ -244,8 +233,7 @@
 	}
 
 	.product-info__label {
-		font-size: rem(16px);
-		font-weight: 700;
+		@include a4a-font-heading-md;
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/marketplace/checkout/style.scss:236]: `font-weight` here is 700, and in the `a4a-font-heading-md` it is 500. Is that ok?

File: client/a8c-for-agencies/sections/overview/body/intro-cards/style.scss [hunk 5]
```diff
@@ -2,15 +2,11 @@
...
-			font-weight: 400;
-			line-height: 24px;
+			@include a4a-font-body-lg;
 		}
 
 		h1 {
-			font-size: rem(32px);
-			font-weight: 700;
-			line-height: 1.25;
+			@include a4a-font-heading-xxl;
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/overview/body/intro-cards/style.scss:9]: font-weight mismatch. Is that ok?

File: client/a8c-for-agencies/style.scss [hunk 6]
```diff
@@ -210,7 +212,8 @@
 	}
 
 	.split-button .button {
-		font-size: 0.875rem;
+		//font-size: 0.875rem;
```
Comment (yashwin) [inline client/a8c-for-agencies/style.scss]: Can we remove this comment?

**Review by yashwin — COMMENTED** [review]
Thanks for working on this 🙇 

This looks great overall 🙌 

I have left a few comments. Let me know what you think. 

> For some reason, the mixins mentioned above didn’t work in the Marketplace section, so I had to duplicate them here: [8516639](https://github.com/Automattic/wp-calypso/commit/8516639144485bd4abc9c9591d10a2e75ddbd7dd) — any reason why the global mixins aren’t usable here?

I removed the code, and it looks like it is working fine. Am I missing something here?

**Comment by keoshi** [comment]
@yashwin Thanks for the thorough review, appreciate it!

Re: `font-weight` comments: yes, I went through them one by one and standardized everything making sure they all looked good. It's going to differ from what we have now, but it's going to create much more consistency across the entire system.

> I removed the code, and it looks like it is working fine. Am I missing something here?

It wasn't working for me for some reason, but if it works than even better! Just to be clear, you're referring to completely removing this file, right? `client/a8c-for-agencies/sections/marketplace/mixins.scss`

File: client/a8c-for-agencies/style.scss (hunk 6 above)
Comment (keoshi) [reply client/a8c-for-agencies/style.scss]: Done in https://github.com/Automattic/wp-calypso/pull/91798/commits/233eec3e4c20aebc5851415aec79cf22fcdbc6c0

**Comment by yashwin** [comment]
> Re: `font-weight` comments: yes, I went through them one by one and standardized everything making sure they all looked good. It's going to differ from what we have now, but it's going to create much more consistency across the entire system.

Awesome!

> It wasn't working for me for some reason, but if it works than even better! Just to be clear, you're referring to completely removing this file, right? `client/a8c-for-agencies/sections/marketplace/mixins.scss`

No, I meant removing the newly added code from that file starting from:

```
// ==========================================================================
// Automattic for Agencies: Typography definitions
//
// See: pfunGA-1zi-p2
// ==========================================================================
```

We will still need this file.

**Comment by keoshi** [comment]
Where's the facepalm animated emoji when you need it? Sorry, @yashwin, of course we still need that file. 😅 

For some reason the Marketplace wasn't taking on the new styles without these local mixins, but I just confirmed it works now. 🤷 

This should now be ready for a final review. Thanks so much!

**Review by yashwin — APPROVED** [review]
Thanks for addressing my comments, @keoshi!

It Looks great to me now! 🚢 

> Where's the facepalm animated emoji when you need it?

🤦 Found it 😆

**Review by Addison-Stavlo — COMMENTED** [review]
This is great! I love the use of mixins here.

On the dotcom side pretty much everything looks 1-1 as before. The one difference I found was in the sites list heading (both full view and collapsed view once a site is selected for overview):

sites-header-text

A small thing but we will probably want the previous version since it matches the other headings we use in that area (domains, plugins, etc.).

**Comment by keoshi** [comment]
Thanks so much for your review @Addison-Stavlo. Fixed the letter-spacing difference for the Dotcom dashboard in https://github.com/Automattic/wp-calypso/pull/91798/commits/ea8ee680810888dfe206452ab93e4ba0905ae8de

Could you take another look, please? Thanks in advance!

**Review by Addison-Stavlo — COMMENTED** [review]
Im still seeing the same difference I noted above. 🤔 

It seems that the font family changed, inspecting the element for that heading text on /sites i see:

BEFORE
Screenshot 2024-06-28 at 2 53 28 PM




AFTER

Screenshot 2024-06-28 at 2 53 47 PM


`"SF Pro Display", sans-serif` is also what Im seeing on the corresponding domains/plugins pages.

**Comment by keoshi** [comment]
> "SF Pro Display", sans-serif is also what Im seeing on the corresponding domains/plugins pages.

That shouldn't be happening in the first place, @Addison-Stavlo. "SF Pro” corresponds to the system font in Apple (`-apple-system`), so we should always use the system font stack defined here:

https://github.com/Automattic/wp-calypso/blob/a1b36280a13857334ca6afd7a3d93ad357f0e053/packages/components/src/styles/_typography.scss#L1

Plus, the “Display” variant of “SF Pro” is deprecated and should not be used.

FYI @lucasmendes-design: we're making the typeface changes here.

**Comment by Addison-Stavlo** [comment]
Interesting @keoshi ! 

Il defer to @lucasmendes-design what we want to do regarding those fonts on the dotcom side. Note this PR makes them inconsistent (changes the one on Sites but not domains/plugins), but we can handle that on our side with a follow up if needed.

**Comment by keoshi** [comment]
> Note this PR makes them inconsistent (changes the one on Sites but not domains/plugins)

This was fixed for Domains in https://github.com/Automattic/wp-calypso/pull/91798/commits/f3bba30af4d20d07f0fc300eead10d27da502d8b and for Plugins in https://github.com/Automattic/wp-calypso/pull/91798/commits/d426e929c896d64c1db6259c24487bc2d55111f2

**Review by Addison-Stavlo — APPROVED** [review]
> This was fixed for Domains in https://github.com/Automattic/wp-calypso/commit/f3bba30af4d20d07f0fc300eead10d27da502d8b and for Plugins in https://github.com/Automattic/wp-calypso/commit/d426e929c896d64c1db6259c24487bc2d55111f2

Awesome, this looks good from my perspective. Thank you!

**Pull request merged**